packages/vaadin-template-renderer/src/vaadin-template-renderer.js

```diff
@@ -1,19 +1,55 @@
 import { FlattenedNodesObserver } from '@polymer/polymer/lib/utils/flattened-nodes-observer.js';
 
 import './vaadin-template-renderer-templatizer.js';
+import './vaadin-template-renderer-grid-templatizer.js';
+
 import { Templatizer } from './vaadin-template-renderer-templatizer.js';
+import { GridTemplatizer } from './vaadin-template-renderer-grid-templatizer.js';
 
-function createRenderer(template) {
-  const templatizer = Templatizer.create(template);
+function createRenderer(component, template, TemplatizerClass = Templatizer) {
+  const templatizer = TemplatizerClass.create(component, template);
 
-  return (root, _owner, model) => {
-    template.__templatizer = templatizer;
-    template.__templatizer.render(root, model);
+  const renderer = (root, _owner, model) => {
+    templatizer.render(root, model);
   };
+
+  template.__templatizer = templatizer;
+  renderer.__templatizer = templatizer;
+
+  return renderer;
+}
+
+function processGridTemplate(component, template) {
+  if (template.matches('.row-details')) {
+    component.rowDetailsRenderer = createRenderer(component, template, GridTemplatizer);
+    return;
+  }
+
+  if (template.matches('.header')) {
+    component.headerRenderer = createRenderer(component, template);
+    return;
+  }
+
+  if (template.matches('.footer')) {
+    component.footerRenderer = createRenderer(component, template);
+    return;
+  }
+
+  if (template.matches('.editor')) {
+    component.editModeRenderer = createRenderer(component, template, GridTemplatizer);
+    return;
+  }
+
+  component.renderer = createRenderer(component, template, GridTemplatizer);
 }
 
 function processTemplate(component, template) {
-  component.renderer = createRenderer(template);
+  if (/^vaadin-grid/.test(component.constructor.is)) {
```

Comment:
Previously, `vaadin-grid` was marked as a runtime dependency in `package.json` to check the component on `instanceof GridElement` here. 

Unfortunately, it was failing visual material tests as it broke the order of styles for `vaadin-grid` when importing `vaadin-grid` in `template-renderer`. More specifically, the `lumo` styles got higher priority than `material` styles.

Reply:
`component.constuctor.is` may not be a reliable check in the long run because:
- the upcoming Lit version will probably not have the `is` property
- someone may create a custom grid extension with a different tag name (`<my-grid>`)

Reply:
Yeah, I agree. Do you have any ideas how to improve that? 

I could import `vaadin-grid`, but then I need to do something with material tests to get the correct order of styles.